### ISSUE TITLE
externalIdField And externalId are same ?

### ISSUE THREAD

**Issue by anilabsinc-ajay** [forcedotcom/SalesforceMobileSDK-iOS#2517]
Please fill out the following details:

1. Version of Mobile SDK Used: 6.1
2. Issue found in Native App or Hybrid App: Native Swift 
3. OS Version: 11
4. Device:All
5. Steps to reproduce:Look into Code
6. Actual behavior:  externalIdField, externalId playing same role
7. Expected Behavior: There should be seprate use of externalIdField
8. Error Log:

 ```
/**
         A factory method for upsert object request.
         ```
         restApi.Promises.upsert(objectType: objectType,externalIdField:"",externalId: "1000", fieldList: {Name: "salesforce.com", TickerSymbol: "CRM"})
         .then { (request) in
            restApi.send(request)
         }
         ```
         - parameters:
             - objectType: Type of object.
             - externalIdField: Identifier for field.
             - fields: Field list as Dictionary.
         - Returns: SFRestRequest wrapped in a promise.
         */
        public func upsert(objectType: String,externalIdField: String, externalId: String, fieldList: Dictionary<String,Any>?) -> Promise<SFRestRequest> {
            return  Promise(.pending) {  resolver in
                resolver.fulfill(self.api!
                    .requestForUpsert(withObjectType: objectType, externalIdField: externalId, externalId: externalId, fields: fieldList!))
            }
        }
        
```

**Comment by trooper2013** [comment]
@anilabsinc-ajay Thanks for reporting. Here is the PR #2518

**Comment by trooper2013** [comment]
Resolved by PR #2518
